Source/WebKit/Platform/IPC/HandleMessage.h

```diff
@@ -294,10 +294,10 @@ void handleMessageSynchronous(StreamServerConnection& connection, Decoder& decod
     static_assert(std::is_same_v<typename ValidationType::CompletionHandlerArguments, typename MessageType::ReplyArguments>);
     using CompletionHandlerType = typename ValidationType::CompletionHandlerType;
 
-    logMessage(connection.connection(), MessageType::name(), object, *arguments);
+    logMessage(connection.protectedConnection(), MessageType::name(), object, *arguments);
```

Comment:
You could add StreamClientConnection::remoteProcessID()
You could move the logging to use the remoteProcessID() directly or make logMessage generic to the connection parameter.

Reply:
I tried this, but as you said IPC::Connection is an implementation details, and `remoteProcessID()` is not present on all platforms, so the code became unwieldy.